### PR TITLE
Make tidy treat "test/ui/foo.nll.stderr" just like "foo.stderr".

### PR DESCRIPTION
**Pull request by pnkfelix**

### CODE REVIEW

**Review by nikomatsakis — APPROVED** [review]

File: src/tools/tidy/src/ui_tests.rs [hunk 1]
```diff
@@ -12,14 +12,41 @@
...
                      &mut |_| false,
                      &mut |file_path| {
         if let Some(ext) = file_path.extension() {
             if (ext == "stderr" || ext == "stdout") && !file_path.with_extension("rs").exists() {
-                println!("Stray file with UI testing output: {:?}", file_path);
-                *bad = true;
+
+                // rust-lang/rust#48879: this fn used to be beautful
+                // because Path API special-cases replacing
+                // extensions. That works great for ".stderr" but not
```
[the remark this answers is not in the document]
Comment (spastorino) [reply src/tools/tidy/src/ui_tests.rs]: beautful -> beautiful

**Comment by pnkfelix** [comment]
closing; we will just fold this into rust-lang/rust#49861

**Pull request closed without merging**